dds/DCPS/DataWriterImpl_T.h

```diff
@@ -301,6 +314,43 @@ DataWriterImpl_T
     Message_Block_Ptr mb;
     ACE_Message_Block* tmp_mb;
 
+    if (marshal_skip_serialize_) {
+      size_t effective_size = 0, padding = 0;
+      if (marshaled_size_) {
+        effective_size = marshaled_size_;
+      } else {
+        TraitsType::gen_find_size(instance_data, effective_size, padding);
+        if (cdr) {
+          effective_size += cdr_header_size;
+        }
+      }
+      if (cdr) {
+        effective_size += padding;
+      }
+      ACE_NEW_MALLOC_RETURN(tmp_mb,
+        static_cast<ACE_Message_Block*>(
+          mb_allocator_->malloc(sizeof(ACE_Message_Block))),
+        ACE_Message_Block(
+          effective_size,
+          ACE_Message_Block::MB_DATA,
+          0, // cont
+          0, // data
+          data_allocator_.get(), // allocator_strategy
+          get_db_lock(), // data block locking_strategy
+          ACE_DEFAULT_MESSAGE_BLOCK_PRIORITY,
+          ACE_Time_Value::zero,
+          ACE_Time_Value::max_time,
+          db_allocator_.get(),
+          mb_allocator_.get()),
+          0);
+      mb.reset(tmp_mb);
+      if(!MarshalTraitsType::to_message_block(*mb, instance_data))
+      {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: DataReaderImpl::dds_demarshal: ")
+                   ACE_TEXT("attempting to skip serialize but bad from_message_block.\n")));
+      }
```

Comment:
similar question about error handling here

Reply:
> similar question about error handling here

The precedent in `dds_marshal` seems to be different than `dds_demarshal`. In `dds_marshal`, I only see error prints and no immediate return. In this case we'd probably want to return a `nullptr` which would likely have other issues in the calling code. I think for now it's good based on the precedent. If you still think it should be different (i.e. return nullptr) then we may want to look at accomplishing that with a separate effort because of expanding scope.